src/utils/filter/getFilter.ts

```diff
@@ -45,17 +49,32 @@ const buildMongoFilter = (filter: any, fields: any[]): any => {
         let startDate: Date;
         let endDate: Date;
         let dateForFilter: any;
+        let startDatetime: Date;
+        let endDatetime: Date;
+        let datetimeForFilter: any;
         switch (field.type) {
           case 'date':
-          case 'datetime':
-          case 'datetime-local':
             dateForFilter = getDateForMongo(value);
-            startDate = dateForFilter.startDate;
-            endDate = dateForFilter.endDate;
+            // startDate represents the beginning of a day
+            startDate = new Date(value);
+            // endDate represents the last moment of the day after startDate
+            endDate = new Date(startDate);
+            endDate.setDate(startDate.getDate() + 1);
+            endDate.setMilliseconds(-1);
+            // you end up with a date range covering exactly the day selected
             value = dateForFilter.date;
```

Comment:
HI @Matthis-M-ReliefApps 
The getFilter utils are used in the filters for the applications, forms, resources pages etc. 
So I think the scope of the ticket only covered src/utils/schema/resolvers/Query/getFilter.ts file, but since the original getFilter's were the same and had the same logic, and there were bugs in the second file too, to maintain consistency between the filters it does make sense to update both (but FYI in the getFilter utils no time filter is used)